src/averageAges.js

```diff
@@ -55,7 +68,20 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(child =>
+    onlyWithSon
+      ? child.sex === 'm' && people.some(mother =>
+        mother.name === child.mother)
+      : people.some(mother => mother.name === child.mother)
```

Comment:
you know what i mean, yeah?
```suggestion
      ? child.sex === 'm' && func(people))
      : func(people))
```